stacks_queues/queue.py

```diff
@@ -15,47 +19,95 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+    
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+        
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1)% self.buffer_size
+        self.size += 1 
 
+    
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
+
         """
-        pass
+
+        #Check if Queue is empty and if so raise Exception
+        # Get the element from the front 
+        # Front will go to the next index in list 
+        # Change size 
+        # Check if the list is empty 
+
+        # deque takes element off the queue, and return the element (Min's comment)
+        
+        if self.size ==0:
+            raise QueueFullException('Queue is empty')
+
+        #in order to deque, you must set self.front to self.front + 1% self.buffer_size
+        deque = self.store[self.front] 
+        self.front = (self.front +1) % self.buffer_size
+        self.size -= 1 
+        return(deque)
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.store[self.front]:
+            return self.front
+        else:
+            return None
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True 
+        else: 
+            return False
```

Comment:
Minor style suggestion to condense your code!
```suggestion
        return self.size == 0
```